src/uct/ib/ud/base/ud_iface.c

```diff
@@ -199,6 +200,22 @@ uct_ud_iface_create_qp(uct_ud_iface_t *self, const uct_ud_iface_config_t *config
     qp_init_attr.cap.max_recv_sge    = 1;
     qp_init_attr.cap.max_inline_data = config->super.tx.min_inline;
 
+    if ((dev->max_inline_data > 0) &&
+        (qp_init_attr.cap.max_inline_data > dev->max_inline_data)) {
```

Comment:
do we really need the first check? I.e.
```suggestion
    if (qp_init_attr.cap.max_inline_data > dev->max_inline_data) {
```

Reply:
yes because currently only EFA populates that value, else it is set to zero.

Reply:
maybe it is better to initialize `dev->max_inline_data` with `INT_MAX` then? (since we have not limit on the device, let's initialize accordingly)

Reply:
fixed, makes sense